astronet/astro_cnn_model/input_ds.py

```diff
@@ -93,6 +95,75 @@ def parse_example(serialized_example):
     ds = tf.data.Dataset.from_tensor_slices(filenames)
     ds = ds.flat_map(tf.data.TFRecordDataset)
     ds = ds.map(parse_example)
+
+###### rough code for checking whether the candidate has been assigned disps E,N,B, or S, and only using these candidates in the data augmentation. There is probably a more concise way to do this ######
+
+
+#     values = np.array([x[1].numpy() for x in ds])
+#    truth_array = []
+#    for array in values:
+#        z = array == [1., 1., 1., 1., 1.]
+#        if z[0] or z[1] or z[3] or z[4]:
+#            truth_array.append(True)
+#        else:
+#            truth_array.append(False)       
+#    truth_data = tf.data.Dataset.from_tensor_slices(truth_array)
+#    ds_new = tf.data.Dataset.zip((ds,truth_data))
+#    new_len = [x for x in ds_new]
+#    print("number trues =",np.array(truth_array).sum())
+#    ds_false = ds_new.filter(lambda image,y:y==False)
+#    false_list = [x for x in ds_false]
+#    randoms = []
+#    for i in range(len(false_list)):
+#        randoms.append(random.randint(0,1))
+#    rand = tf.data.Dataset.from_tensor_slices(randoms)
+#    false_rand = tf.data.Dataset.zip((ds_false,rand))
+#    false_rand = false_rand.filter(lambda x,z: z==1)
+#    ds_new = ds_new.filter(lambda image,y: y ==True)
+    
+    features = list(configurations.final_alpha_0()['inputs']['features'].keys())
+    globals = [feature for feature in features if 'global' in feature]
+    locals = [feature for feature in features if 'local' in feature]
+    secondaries = [feature for feature in features if 'secondary' in feature]
+    all = globals+locals+secondaries
+
+
+    def add_noise(image,y):
+        for a in array_noise:
+            z = image[0][a]
+            noise = 1.48*stats.median_absolute_deviation(z, scale=1)
```

Comment:
I think we want to add `nan_policy='omit'` since we may have series that have a few nans in them

Reply:
I think also we still want to do `tf.random.normal` sample? otherwise we'll just return `z` shifted up by the same `noise` at every position?